### PR TITLE
fix: wrong schema id is stored for credentials

### PR DESCRIPTION
**Pull request by pallavighule**
### What  ###

- Fixed the issue that wrong schema id is stored for credentials
- Written a test case to register and resolve a credential definition (internal,credDefDid != schemaDid)

### CODE REVIEW

**Comment by pallavighule** [comment]
@TimoGlastra @genaris  I have fixed the  issue #1874  .Please review the PR.

File: packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts [hunk 1]
```diff
@@ -979,8 +980,8 @@ export class IndyVdrAnonCredsRegistry implements AnonCredsRegistry {
     }
 
     const schema = response.result.data?.txn.data as SchemaType
-
-    const schemaId = getUnqualifiedSchemaId(did, schema.data.name, schema.data.version)
+    const schemaDid = response.result.data?.txn.metadata.from as string
+    const schemaId = getUnqualifiedSchemaId(schemaDid, schema.data.name, schema.data.version)
 
     return {
       schema: {
```
Comment (genaris) [inline packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts:987]: I understand from this code that the actual issuer Id for this schema is the one in the "from" field. So shouldn't we also put this value into the `issuerId` of the returned object?

File: packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts [hunk 2]
```diff
@@ -287,8 +287,9 @@ export class IndyVdrAnonCredsRegistry implements AnonCredsRegistry {
       }
 
       // Format the schema id based on the type of the credential definition id
+      const unqualifiedDid = schema.schema.schemaId.split(':')[0]
       const schemaId = credentialDefinitionId.startsWith('did:indy')
-        ? getDidIndySchemaId(pool.indyNamespace, namespaceIdentifier, schema.schema.name, schema.schema.version)
+        ? getDidIndySchemaId(pool.indyNamespace, unqualifiedDid, schema.schema.name, schema.schema.version)
```
Comment (genaris) [inline packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts:292]: If my comment below regarding returning the correct issuerId in `fetchIndySchemaWithSeqNo` is correct, I think we could simply `schema.schema.issuerId`, so there will not be need of creating the unqualifiedDid variable.

**Review by genaris — COMMENTED** [review]
Thanks! I have a couple questions regarding the returned value of `fetchIndySchemaWithSeqNo` that could make the code even more simple (if I'm right).

File: packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts (hunk 1 above)
Comment (pallavighule) [reply packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts:987]: Yes, thank you for the suggestion. I will make the necessary changes.

File: packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts (hunk 2 above)
Comment (pallavighule) [reply packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts:292]: Yes right , I will make the necessary changes.

File: packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts (hunk 1 above)
Comment (pallavighule) [reply packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts:986]: @genaris  I have made the necessary changes

File: packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts (hunk 2 above)
Comment (pallavighule) [reply packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts]: I have made the necessary changes

**Review by genaris — APPROVED** [review]
LGTM!

It would be good to get a double check by @KolbyRKunz (who also experienced this issue recently)

**Comment by KolbyRKunz** [comment]
LGTM!

**Pull request merged**